src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3448,9 +3448,17 @@ public long GetIndexEtag(QueryOperationContext context, QueryMetadata q)
                 using (_contextPool.AllocateOperationContext(out TransactionOperationContext indexContext))
                 {
                     using (indexContext.OpenReadTransaction())
-                    using (context.OpenReadTransaction())
```

Comment:
I don ot like this change. You should use 'CalculateIndexEtag' directly. IIRC the order of opened transactions heree mattered.